src/util.rs

```diff
@@ -148,7 +148,7 @@ fn handle_signal(signal: Completed) -> Result<zvariant::OwnedValue, Error> {
     if args.dismissed {
         Err(Error::Prompt)
     } else {
-        Ok(args.result.into())
+        Ok(zvariant::OwnedValue::try_from(args.result)?)
```

Comment:
Question: Can we do `.map_err(From::from)` or similar here instead to avoid the `Ok(?)` pattern?

Reply:
Seems to work, changed.